client/lib/products-values/index.js

```diff
@@ -249,6 +253,13 @@ export function isDomainTransfer( product ) {
 	return isTransfer( product );
 }
 
+export function isDomainTransferPrivacy( product ) {
+	product = formatProduct( product );
+	assertValidProduct( product );
+
+	return product.product_slug === 'domain_transfer_privacy';
```

Comment:
This should be the `TRANSFER_IN_PRIVACY` constant from `lib/domains/constants.js`.